File: 프로그래머스/Lv.3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    queue = deque()
    visited = [False] * n
    
    for computer in range(n):
        if visited[computer] == False:
            queue.append(computer)
            answer += 1
            while queue:
                temp = queue.popleft()
                for connect in range(n):
                    if temp != connect and computers[temp][connect] == 1 and visited[connect] == False:
                        visited[connect] = True
                        queue.append(connect)
        
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    
    const visited = Array(n).fill(false);
    for (let index = 0; index < n; index++) {
        if (!visited[index]) {
            answer += 1;
            
            const deque = [index];
            while (deque.length > 0) {
                const currentNode = deque.shift();
                
                for (let nodeIndex = 0; nodeIndex < n; nodeIndex++) {
                    if (!visited[nodeIndex] && currentNode != nodeIndex && computers[currentNode][nodeIndex] === 1) {
                        deque.push(nodeIndex);
                        visited[nodeIndex] = true;
                    }
                }
            }
        }
    }
    
    return answer;
}